zebra-network/src/peer/connection.rs

```diff
@@ -843,6 +829,25 @@ where
                         Either::Right((Some(Ok(peer_msg)), _cancel)) => {
                             self.update_state_metrics(format!("Out::Rsp::{}", peer_msg.command()));
 
+                            // `addr` responses are limited to avoid peer set takeover. Any excess
+                            // addresses are stored in `cached_addrs`.
+                            let mut with_cached_addrs = |handler: &Handler, msg| {
+                                if let (Handler::Peers, Message::Addr(new_addrs)) = (handler, &msg)
+                                {
+                                    // Security: This method performs security-sensitive operations, see its comments
+                                    // for details.
+                                    let response_addrs = Handler::update_addr_cache(
+                                        &mut self.cached_addrs,
+                                        new_addrs,
+                                        PEER_ADDR_RESPONSE_LIMIT,
+                                    );
+
+                                    Message::Addr(response_addrs)
+                                } else {
+                                    msg
+                                }
+                            };
```

Comment:
Maintainability Issue: Putting this closure in the middle of a 300 line method makes both that method and the closure harder to read.